types/module/module.go

```diff
@@ -185,13 +185,19 @@ func (bm BasicManager) AddQueryCommands(rootQueryCmd *cobra.Command) {
 // AppModuleGenesis is the standard form for an application module genesis functions
 type AppModuleGenesis interface {
 	AppModuleBasic
-	HasGenesis
+	HasGenesisBasics
 }
 
 // HasGenesis is the extension interface for stateful genesis methods.
 type HasGenesis interface {
 	HasGenesisBasics
-	InitGenesis(sdk.Context, codec.JSONCodec, json.RawMessage) []abci.ValidatorUpdate
+	InitGenesis(sdk.Context, codec.JSONCodec, json.RawMessage)
+	ExportGenesis(sdk.Context, codec.JSONCodec) json.RawMessage
+}
+
+type HasABCIGenesis interface {
+	HasGenesisBasics
+	InitABCIGenesis(sdk.Context, codec.JSONCodec, json.RawMessage) []abci.ValidatorUpdate
```

Comment:
Can we maybe still call that method InitGenesis ? So that it makes sure a module can implement only of those interfaces?